t/90h2olog.t

```diff
@@ -0,0 +1,92 @@
+#!perl
+# DTRACE_TEST=1 to skip to check prereqisites except for OS
+# H2OLOG_DEBUG=1 for more runtime logs
+use strict;
+use warnings FATAL => "all";
+use Net::EmptyPort qw(empty_port);
+use Test::More;
+use JSON;
+use t::Util;
+
+plan skip_all => "h2olog is supported only for Linux"
+    if $^O ne 'linux';
```

Comment:
> assuming that the intent of the environment variable is to skip any dependency check.

Yes, its intent is not to skip this test if `DTRACE_TEST=1` is given. However, there's a case to set `DTRACE_TEST=1` on macOS, where this test should be skipped.

Reply:
Could you please pinpoint to the line that set `DTRACE_TESTS` for macOS? I cannot find that.

Separately, to me it seems that we should move the requirement checks inside `unless` below outside of that `unless`. The reason is that the tests would never succeed because the prerequisites are not met.

I think our principle should be:
* Skip tests when prerequisites are not met (because the tests would always fail).
* When a test is expected to fail in general (e.g., an external dependency requires a special patch; consider the case of bpftrace), and if it is hard to tell if the workaround has been deployed, skip the test by default and use an environment variable to turn on the test.
* When a test is expected to succeed in general but might fail in some deployments (e.g., due to being run on some awkward platform), provide a knob to skip that test.

Do these sound reasonable? Does `DTRACE_TESTS` fit into the second category? Which of the two exceptions do we want to use for h2olog?

Reply:
> Could you please pinpoint to the line that set DTRACE_TESTS for macOS? I cannot find that.

I've assumed a local macOS development. It's also possible if we add a macOS flavor to CI, although we have no concrete plan to do so. 

> our principle

I think we are on the same page for the principal.

However, I have concerns about the case where the h2olog tests are skipped when we expect them to fail. The h2olog tests have a lot of prerequisites, and the prerequisites are easily dropped by mistake, leading to the tests being skipped. For example, I made a regression (https://github.com/h2o/h2o/pull/2546/files) that the `dtrace` CI flavor was misconfigured, where the dedicated Docker container was not set correctly, but dtrace-related tests did not fail; they were just skipped. 

I don't want to violate the principle, but I'd like to force enabling the h2olog tests on particular CI flavors. Any ideas?

Reply:
> I don't want to violate the principle, but I'd like to force enabling the h2olog tests on particular CI flavors. Any ideas?

I agree that it is good to have a way to force the execution of certain tests.

My point is that this line goes against that idea, because it skips all the tests without consulting the environment variable. Therefore I am suggesting to move this skip_all-if thing inside the `unless` block below.

Reply:
Ah, yes, it makes sense. It means we believe those who `DTRACE_TESTS=1`  know what they are doing. 

4bc280d761f13ec1a3314de9433487694ede8363

BTW I just removed the OS check, which is duplicated in `-e $h2olog_prog`.